server/webapp/WEB-INF/rails.new/app/views/admin/tasks/fetch/_form.html.erb

```diff
@@ -1,50 +1,231 @@
     </div>
+  <% else %>
+    <div class="fetch-artifact-config">
+      <%= render :partial => 'admin/tasks/fetch/gocd_artifact_config', :locals => {:scope => {:var_form => scope[:form], :task => scope[:task]}} %>
+    </div>
+  <% end %>
 </div>
 
 <script type="text/javascript">
-    Util.on_load(function() {
-        var pipelines = <%== @pipeline_json %>;
-        var currentPipelineName = '<%= @pipeline.name() %>';
-        var fetchTaskEditor = new TaskEditor.FetchTaskEditor();
-        fetchTaskEditor.wireInAutocomplete(jQuery("#task_pipelineName"), jQuery("#task_stage"), jQuery("#task_job"), pipelines, currentPipelineName);
+  var artifactIdToPluginId    = <%== com.google.gson.Gson.new.toJson(scope[:artifact_id_to_plugin_id]) %>;
+  var pluginToFetchView       = <%== com.google.gson.Gson.new.toJson(scope[:artifact_plugin_to_fetch_view]) %>;
+  var routeFetchTask          = "<%== scope[:routeFetch] %>";
+  var routeFetchPluggableTask = "<%== scope[:routeFetchPluggable] %>";
+  var currentPipelineName     = '<%= @pipeline.name() %>';
+  var isFormEdited            = false;
+
+  Util.on_load(function () {
+    var pipelines       = <%== @pipeline_json %>;
+    var fetchTaskEditor = new TaskEditor.FetchTaskEditor();
+    fetchTaskEditor.wireInAutocomplete(jQuery("#task_pipelineName"), jQuery("#task_stage"), jQuery("#task_job"), pipelines, currentPipelineName);
+    jQuery('#task_form input[type="radio"]').on('change', showFetchTaskView);
+    jQuery('#task_pipelineName, #task_stage, #task_job').on('input propertychange paste result', function () {
+      showFetchTaskView(true);
+    });
+
+    showFetchTaskView(false);
+  });
+
+  function showFetchTaskView(isEdited) {
+    isFormEdited = isEdited;
+    getSelectedTaskType() === "external" ? renderExternalFetchTaskView() : renderGoCDFetchTaskView();
+  }
+
+  function renderGoCDFetchTaskView() {
+    jQuery("#task_form").attr("action", routeFetchTask);
+    jQuery(".fetch-artifact-config").html(getTemplate("gocd-fetch-artifact_template"));
+  }
+
+  function renderExternalFetchTaskView() {
+    jQuery("#task_form").attr("action", routeFetchPluggableTask);
+    jQuery(".fetch-artifact-config").html(getTemplate("external-fetch-artifact_template"));
+    clearArtifactId();
+
+    if (isAnyParameterized('task_pipelineName', 'task_stage', 'task_job')) {
+      showPluginSelectionDropdown();
+    } else {
+      hidePluginSelectionDropdown();
+      initArtifactIdAutoComplete();
+      jQuery("#task_artifactId").on("input propertychange paste result", renderViewForSelectedArtifactId)
+    }
+  }
+
+  function clearArtifactId() {
+    if (isFormEdited) {
+      jQuery("#task_artifactId").val("");
+    }
+  }
+
+  function renderViewForSelectedArtifactId() {
+    var artifactId = jQuery("#task_artifactId").val();
+    if (!artifactId || artifactId.trim().length === 0) {
+      hidePluginSelectionDropdown();
+      return;
+    }
+
+    if (artifactId && artifactId.startsWith("#{")) {
```

Comment:
Make it a contains check? Because I can specify the artifactId as `foo-#{bar}` too.